linera-core/src/chain_worker/state/attempted_changes.rs

```diff
@@ -298,33 +297,30 @@ where
         );
 
         let required_blob_ids = executed_block.required_blob_ids();
-        // Verify that no unrelated blobs were provided.
-        self.state
-            .check_for_unneeded_blobs(&required_blob_ids, blobs)?;
-        let blobs = self.state.get_required_blobs(executed_block, blobs).await?;
-        let blobs = blobs.into_values().collect::<Vec<_>>();
-
-        let certificate_hash = certificate.hash();
+        let blobs_result = self.state.get_required_blobs(executed_block, &[]).await;
+        let blobs_result = blobs_result.map(|blobs| blobs.into_values().collect::<Vec<_>>());
```

Comment:
Can we chain the two lines instead of creating a variable with the same name? So
```rust
self.state.get_required_blobs(executed_block, &[]).await.map(|blobs| ...)
```